src/msspreadsheets/spreadsheet_endpoint.bal

```diff
@@ -0,0 +1,639 @@
+            return tableError;
+        }
+    }
+
+    # Open a Table.
+    # + tableName - name of the table to be opened
+    # + return - A Table client object on success, else returns an error
+    public remote function openTable(string tableName) returns @tainted Table|error {
+        //Make a GET request and retrieve the table information
+        http:Request request = new;
+        http:Response|error response = self.worksheetClient->get(<@untainted> ("/v1.0/me/drive/root:" + self.properties.path +
+                                            self.properties.workbookName + ".xlsx:/workbook/worksheets/" + 
+                                            self.properties.worksheetName + "/tables/" + tableName), request);
+
+        if response is error {
+            HttpError httpError = error(HTTP_ERROR, message = "Error occurred while accessing the Microsoft Graph API.", 
+                                        errorCode = HTTP_ERROR, cause = response);
+            return httpError;
+        }
+
+        http:Response httpResponse = <http:Response> response;
+
+        if httpResponse.statusCode != http:STATUS_OK {
+            HttpResponseHandlingError httpResponseHandlingError = error(HTTP_RESPONSE_HANDLING_ERROR,
+                message = "Error occurred while inserting data into table.", errorCode = HTTP_RESPONSE_HANDLING_ERROR);
+            return httpResponseHandlingError;
+        }
+
+        //If the table exists we will get a JSON response with the table's information
+        json|error responseJson = httpResponse.getJsonPayload();
+
+        if !(responseJson is map<json>) {
+            typedesc<any|error> typeOfRespone = typeof responseJson;
+            TypeConversionError typeError = error(TYPE_CONVERSION_ERROR, 
+                    message = "Invalid response; expected a `map<json>` found " +  typeOfRespone.toString(), 
+                    errorCode = TYPE_CONVERSION_ERROR);
+            return typeError;
+        }
+
+        map<json> payload = <map<json>> responseJson;
+    
+        json|error identifier = payload.id;
+        if !(identifier is string) {
+            typedesc<any|error> typeOfIdentifier = typeof identifier;
+            TypeConversionError typeError = error(TYPE_CONVERSION_ERROR, 
+                    message = "Invalid identifier; expected a `string` found " +  typeOfIdentifier.toString(), 
+                    errorCode = TYPE_CONVERSION_ERROR);
+            return typeError;
+        }
+
+        string sheetIdentifier = <string> identifier;
+
+        //Address is not returned from the above API call. Hence the address is initialized to an empty string
+        string address = "";
+
+        Table resultsTable = <@untainted> new (self.microsoftGraphConfig, self.properties.path, 
+                    self.properties.workbookName, self.properties.sheetId, self.properties.worksheetName,
+                    sheetIdentifier, address, tableName);
+
```

Comment:
```suggestion
```

Reply:
I think we should have a space between these two statements.

```
        Table resultsTable = <@untainted>new (self.microsoftGraphConfig, self.properties.path,
            self.properties.workbookName, self.properties.sheetId, self.properties.worksheetName,
            sheetIdentifier, address, tableName);

        return resultsTable;
```